apps/users/forms/registration.py

```diff
@@ -0,0 +1,110 @@
+from crispy_forms.helper import FormHelper
+from crispy_forms.layout import Submit
+from django import forms
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from apps.users.models.user.constants import NATIONALITIES
+from apps.users.models.user.institution import InstitutionFontys, InstitutionTue
+from apps.users.models.user.profile import Profile
+
+
+class RegisterUserForm(UserCreationForm):
+
+    first_name = forms.CharField(max_length=30, required=True, initial="")
+    last_name = forms.CharField(max_length=30, required=True, initial="")
+    username = forms.EmailField(
+        max_length=254,
+        label="Institutional Email",
+        help_text="Required. Please fill in your institutional email address.",
+        required=True,
+        initial="",
+    )
+    email = forms.EmailField(
+        max_length=254,
+        label="Personal email",
+        required=False,
+        help_text="Optional, Please fill in a valid email address.",
+        initial="",
+    )
+    password1 = forms.CharField(widget=forms.PasswordInput, required=True, initial="")
+    password2 = forms.CharField(widget=forms.PasswordInput, required=True, initial="")
+
+    # profile fields
+    nationality = forms.ChoiceField(choices=list(zip(NATIONALITIES, NATIONALITIES)))
+    terms_confirmed = forms.BooleanField()
+    subscribed_newsletter = forms.BooleanField(required=False)
+
+    class Meta:
+        model = User
+        fields = ["first_name", "last_name", "username", "email", "password1", "password2"]
+
+    def clean_nationality(self):
+        data = self.cleaned_data["nationality"]
+        if data not in NATIONALITIES:
+            raise ValidationError("Please enter your nationality.")
+        return data
+
+    def clean_username(self):
+        data = self.cleaned_data["username"]
+        if not (data.endswith("@student.tue.nl") or data.endswith("@alumni.tue.nl") or data.endswith("@fontys.nl")):
```

Comment:
we need to consolidate this check (maybe in the model?) as it's used in other places (at the moment, only UpdateProfileForm and RegisterUserForm)